driver-mapping/src/main/java/com/datastax/driver/mapping/configuration/naming/CommonNamingConventions.java

```diff
@@ -0,0 +1,296 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping.configuration.naming;
+
+import java.util.*;
+
+/**
+ * Implementations of industry common naming conventions
+ */
+public class CommonNamingConventions {
+
+    /**
+     * E.g. myXmlParser / myXMLParser
+     * When constructing, upperCaseAbbreviations may be used to decide:
+     * <ul>
+     * <li>{@code false} - myXmlParser (default)</li>
+     * <li>{@code true} - myXMLParser</li>
+     * </ul>
+     */
+    public static class LowerCamelCase extends CamelCase {
```

Comment:
More documentation on these classes would be appreciated, and also on the constructors as some of them aren't clear to me (i.e. It isn't immediately clear to me what `ignorablePrefixes` means)